app/controllers/floors_controller.rb

```diff
@@ -104,9 +104,13 @@ def find_or_create_label(location, params)
   end
 
   def save_location_on_icon(location, icon_id)
-    icon = Icon.find(icon_id)
-    icon.locations << location
-    icon.save
+    return location.icon unless location.icon.nil?
+    icon = nil
+    if !icon_id.blank?
```

Comment:
if this is still the default string which is an empty string

Reply:
```
if Icon.exists?(icon_id)
  ...
end
```